File: sdks/typescript/src/providers/anthropic.ts
import { provedMessage } from "./anthropic_like.js";
import type {
  JsonObject,
  ProofLayerAttachment,
  ProviderCaptureOptions
} from "../types.js";
import type { ProofLayer } from "../proof_layer.js";

export function withProofLayer(
  client: {
    messages?: { create?: (params: JsonObject) => Promise<JsonObject> };
  },
  proofLayer: ProofLayer,
  captureOptions: ProviderCaptureOptions = {}
): {
  messages: {
    create: (params: JsonObject) => Promise<JsonObject & { proofLayer: ProofLayerAttachment }>;
  };
} {
  return {
    messages: {
      create: async (params: JsonObject) => {
        const result = await provedMessage(client, params, proofLayer, captureOptions);
        return {
          ...result.message,
          proofLayer: {
            bundleId: result.bundleId,
            bundleRoot: result.bundleRoot,
            signature: result.signature,
            createdAt: result.createdAt,
            bundle: result.bundle
          }
        };
      }
    }
  };
}

File: packages/sdk-python/proofsdk/providers/anthropic.py
from __future__ import annotations

from typing import Any

from proofsdk.providers.anthropic_like import proved_message


def with_proof_layer(client, proof_layer, capture_options: dict[str, Any] | None = None):
    capture_options = capture_options or {}

    class _WrappedMessages:
        def create(self, params: dict[str, Any]) -> dict[str, Any]:
            message, proof = proved_message(
                client.messages.create,
                params,
                proof_layer,
                capture_options=capture_options,
            )
            return {
                **message,
                "proof_layer": {
                    "bundle_id": proof["bundle_id"],
                    "bundle_root": proof["bundle_root"],
                    "signature": proof["signature"],
                    "created_at": proof.get("created_at"),
                    "bundle": proof.get("bundle"),
                },
            }

    class _WrappedClient:
        messages = _WrappedMessages()

    return _WrappedClient()
